### PR TITLE
Code for the Old God! Old Cult is back!

### PR DESCRIPTION
**Pull request by Robustin**
Old Cult has returned to its last form prior to removal,  with just a few tweaks:
- The "X acolytes must escape" objective has been disabled.
- Invoking the imbue talisman will consume the imbue talisman (RIP talisman factories)
- Invoking a stun talisman will brute the caster for 15, up from 10
- Stun talisman's stun duration reduced to 9 tics, down from 10
- Stun talisman no longer mutes for the full duration, now applies 4 tics of mute, down from 10, and 15 tics of stutter and a _new_ cultslur proc that should make it difficult to decipher radio messages for help, but with the downside of making it obvious that the victim isn't merely drunk
- Cult retains the timestop rune, requiring 3 casters (fun rune, can't be as OP as blood boil anyway)
- Experimental: Drawing a Nar-Sie rune will send out a notification similar to newcult's. Drawing the nar-sie rune now takes 10x the time of a normal rune. If coded properly (hah) it should name the cultist who is drawing the rune and their location, giving the crew a small window to intervene before the ritual is complete (invoking the rune is still the same and will occur instantly).

### CODE REVIEW

File: code/game/gamemodes/cult/cult.dm [hunk 1]
```diff
@@ -59,8 +82,8 @@
 		restricted_jobs += "Assistant"
 
 	//cult scaling goes here
-	recommended_enemies = 1 + round(num_players()/8)
-	orbs_needed = recommended_enemies
+	recommended_enemies = 3 + round(num_players()/20)
```
Comment (KorPhaeron) [inline code/game/gamemodes/cult/cult.dm]: >scaling starting cultists when their entire objective is to have X people and the round ends

At least it's a bit more sane than the previous oldcult scaling that would routinely give you 8-9 people I guess

**Comment by KazeEspada** [comment]
Dont remove new cult. Do both as seperate gamemodes.

File: code/modules/power/singularity/narsie.dm [hunk 2]
```diff
@@ -61,7 +61,7 @@
 	return
 
 /obj/singularity/narsie/proc/godsmack(atom/A)
-	if(isobj(A))
+	if(istype(A,/obj/))
 		var/obj/O = A
```
Comment (KorPhaeron) [inline code/modules/power/singularity/narsie.dm]: Why'd you change this back?

**Comment by KazeEspada** [comment]
Also summon nar nar should always be an objective. And nar nar consuming your target is a sac.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 3]
```diff
@@ -183,10 +183,13 @@
...
 		M.stuttering += 4
 		M.Dizzy(5)
+		if(iscultist(M) && prob(5))
+-			M.say(pick("Av'te Nar'sie","Pa'lid Mors","INO INO ORA ANA","SAT ANA!","Daim'niodeis Arc'iai Le'eones","R'ge Na'sie","Diabo us Vo'iscum","Eld' Mon Nobis"))
 	if(data >= 75 && prob(33))	// 30 units, 135 seconds
 		if (!M.confused) M.confused = 1
 		M.confused += 3
-		if((is_handofgod_cultist(M) && !is_handofgod_prophet(M)))
+		if(iscultist(M) || (is_handofgod_cultist(M) && !is_handofgod_prophet(M)))
+			ticker.mode.remove_cultist(M.mind)
```
Comment (KorPhaeron) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: I think security mass arrest+deconvert is dull gameplay personally.

If we gotta have conversion back I'd at least prefer no deconversion.

File: html/changelogs/Robustin-Oldcult [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+# Also, this gets changed to [] after reading.  Just remove the brackets when you add new shit.
+# Please surround your changes in  double quotes ("). It works without them, but if you use certain characters it screws up compiling. The quotes will not show up in the changelog.
+changes: 
+  - rscadd: "Old Cult is back baby!"
+  - tweak: "Cult survival objective disabled"
+  - tweak: "Imbuing talismans now consumes the imbue rune"
+  - tweak: "Invoking the stun talisman now hurts the cultist more"
+  - tweak: "The stun talisman duration is nerfed by 10%, mute is nerfed by 60% but a stutter and special new slur effect will last much longer"
+  - tweak: "The old cult has retained the rune of time stop, now requiring 3 cultists to invoke"
+  - rscdel: "New Cult is removed!"
```
Comment (KorPhaeron) [inline html/changelogs/Robustin-Oldcult]: In the future you can do changelogs from the PR body

File: code/modules/power/singularity/narsie.dm (hunk 2 above)
Comment (duncathan) [inline code/modules/power/singularity/narsie.dm]: He probably doesn't know what any of it does

File: code/game/gamemodes/cult/cult.dm (hunk 1 above)
Comment (Robustin) [inline code/game/gamemodes/cult/cult.dm]: The "survive" objective is gone.

Comment (KorPhaeron) [inline code/game/gamemodes/cult/cult.dm]: Yeah but the nar-sie stand in a square with 9 guys isn't

File: code/modules/power/singularity/narsie.dm (hunk 2 above)
Comment (Robustin) [inline code/modules/power/singularity/narsie.dm]: Ding ding ding @duncathan gets it.

I'll change this back if doesn't screw with oldcult.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 3 above)
Comment (Robustin) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: The deconversion takes more effort than almost any other. You have to babysit the cultists and keep them separated since a single loose cultist with a tome can undo everything. You also also have to babysit if you want to confirm that they've been deconverted since theres only a brief moment when you can tell that a cultist is purified. 

Also being able to summon cultists during this process makes it super risky as opposed to shooting them on the spot.

**Comment by Robustin** [comment]
The dreaded build failure. Still have no idea how to narrow down what the problem is. Changed a lot of lines and I'm sure there's some human error in there.

**Comment by KazeEspada** [comment]
code/game/gamemodes/cult/cult.dm:31:error: bad argument definition
code/game/gamemodes/cult/cult.dm:33:error: bad argument definition
code/game/gamemodes/cult/cult.dm:43:error: inconsistent indentation
code/game/gamemodes/cult/cult.dm:45:error: inconsistent indentation
code/game/gamemodes/cult/cult.dm:44:warning: empty 'else' clause
code/game/gamemodes/cult/cult.dm:50:error: bad argument definition
code/game/gamemodes/cult/ritual.dm:142:error: .: expected end of statement
code/game/gamemodes/cult/ritual.dm:142:error: missing left-hand argument to <.

**Comment by Razharas** [comment]
@Robustin 
Maybe you should try to finish your carp karate pr first?
THis level of coding seem above your current coding powerlevel

**Comment by KazeEspada** [comment]
Gamemode code is a monstrosity(which is why no one reverted cult already).

**Comment by ChangelingRain** [comment]
Get in coderbus so we can yell at you okay

**Comment by Buggy123** [comment]
:+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1:
>50% of the server has been begging for this, thank you.

**Comment by optimumtact** [comment]
Why was this closed?

**Comment by xxalpha** [comment]
Because memetainers.

File: html/changelogs/Robustin-Oldcult (hunk 4 above)
Comment (optimumtact) [inline html/changelogs/Robustin-Oldcult]: nothing wrong with this way though

**Comment by PKPenguin321** [comment]
Not the solution to stun talismans that I would use personally, but :+1:

**Comment by ChangelingRain** [comment]
Cheridan said I could.
That, and I want robustin to get into coderbus.

File: code/game/gamemodes/cult/talisman.dm [hunk 5]
```diff
@@ -208,6 +244,39 @@ Rite of Disorientation
...
 
+/obj/item/weapon/paper/talisman/stun/attack_self(mob/living/user)
+	if(iscultist(user))
+		user << "<span class='warning'>To use this talisman, attack the target directly.</span>"
+	else
+		user << "<span class='danger'>There are indecipherable images scrawled on the paper in what looks to be... <i>blood?</i></span>"
+		
+/obj/item/weapon/paper/talisman/stun/attack(mob/living/target, mob/living/user)
+	if(iscultist(user))
+-		user.whisper(invocation)
```
Comment (Shadowlight213) [inline code/game/gamemodes/cult/talisman.dm]: More extra minuses.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 6]
```diff
@@ -183,10 +183,13 @@
 		if (!M.stuttering) M.stuttering = 1
 		M.stuttering += 4
 		M.Dizzy(5)
+		if(iscultist(M) && prob(5))
+-			M.say(pick("Av'te Nar'sie","Pa'lid Mors","INO INO ORA ANA","SAT ANA!","Daim'niodeis Arc'iai Le'eones","R'ge Na'sie","Diabo us Vo'iscum","Eld' Mon Nobis"))
```
Comment (Shadowlight213) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: Remove extra minus

File: code/game/gamemodes/wizard/soulstone.dm [hunk 7]
```diff
@@ -123,6 +123,12 @@
 
 		if("VICTIM")
 			var/mob/living/carbon/human/T = target
+			if(ticker.mode.name == "cult" && T.mind == ticker.mode:sacrifice_target)
+-				if(iscultist(user))
```
Comment (Shadowlight213) [inline code/game/gamemodes/wizard/soulstone.dm]: Delete extra minuses

**Comment by Robustin** [comment]
Made all the fixes to the code, still waiting for @ChangelingRain to re-open. Literally dragged onto IRC so Cheridan and Joan can whine about stun talismans and Nar-Sie as an objective when this PR addresses both in a way the should improve oldcult.

**Comment by ChangelingRain** [comment]
Robustin has sat around in coderbus for a while.

**Comment by Robustin** [comment]
Why in the world am I getting Travis Errors on files I haven't even touched. Inconsistent Indentation on Human.dm?

Plz

File: code/modules/mob/living/carbon/human/human.dm [hunk 8]
```diff
@@ -706,7 +706,7 @@
 
 /mob/living/carbon/human/singularity_pull(S, current_size)
 	if(current_size >= STAGE_THREE)
-		var/list/handlist = list(l_hand, r_hand)
+	var/list/handlist = list(l_hand, r_hand)
```
Comment (Shadowlight213) [inline code/modules/mob/living/carbon/human/human.dm]: You accidentally unidentified this. That's the source of your error

Comment (Robustin) [inline code/modules/mob/living/carbon/human/human.dm]: I intentionally did it to see if I can't get rid of whatever first triggered that error.

**Comment by duncathan** [comment]
Cold for the Old Gold! Old Cold is Bold!

**Comment by Zilenan91** [comment]
The main reason oldcult was removed in the past was because stunpaper was ultimately too effective. I still say the best place for it would be in supply talismans only to carry cultists through the early game and nothing more.

**Comment by duncathan** [comment]
Actually zilenan the main reason old cult was removed was because your incessant whining actually convinced us that people hated it

**Comment by Zilenan91** [comment]
More than a little of that yeah.

**Comment by drynwyn** [comment]
holy shit did zile"Whining about something constantly until it's changed is the only way to do anything here"nan just acknowledged fault

**Comment by Zilenan91** [comment]
I've literally never once made it a secret that I've done these things.

**Comment by Core0verload** [comment]
Fuck yeah. :+1: 

The only thing i'm concerned about is communication runes. Do they still have multiple uses or do they disappear after the first one?

**Comment by Robustin** [comment]
Communication rune functionality was moved to the tomes a while ago. You can communicate using your tome or (if you lose it) you can damage yourself to send out a "flawed" message. 

Still not sure how using the web editor has pissed off a random line of some random code I didn't touch but I'm poking around for answers.

**Comment by Steelpoint** [comment]
If Oldman can get this to work then I support it and I may turn Cultist back on as a antag option.

Also I see no problem with allowing Sec to deconvert a cultist, as he noted its a massive timesink and has a high resource requirment that needs you to babysit that one person for several minutes, and its risky to the point you're better off executing them, but the reward is a new loyal crewmember.

**Comment by ShadowDeath6** [comment]
The inglorious return of stun talismans
And the glorious return of everything else
I missed oldcult :+1:

File: code/modules/mob/mob_defines.dm [hunk 9]
```diff
@@ -79,6 +79,7 @@
 	var/obj/item/r_hand = null//Living
 	var/obj/item/weapon/storage/s_active = null//Carbon
 
+	var/seer = 0 //for cult//Carbon, probably Human
```
Comment (Shadowlight213) [inline code/modules/mob/mob_defines.dm]: Where is this actually used? Just use see_ov

Comment (Robustin) [inline code/modules/mob/mob_defines.dm]: One of the many lines where I said "why does this exist?" and answered "I dont know, better add it anyway".

**Comment by Shadowlight213** [comment]
@Robustin I downloaded your pr and am working on fixing all the errors. Gimme a bit and I'll PR it to your branch.

**Comment by Robustin** [comment]
Hah, I'd appreciate it. I haven't touched code in ~10 years but I never remember C++/Java being this bitchy about indentation. 

I think I got most of the cult.dm stuff.

**Comment by RemieRichards** [comment]
@Robustin 
Most C family languages use {} for deciding where a block of code starts or ends.
Other Languages use Whitespace, the level of indentation to decide where a block starts or end.
While you can indent in C languages (and its strongly encouraged) its purely for readability there.

Byond is sort of a C family language, and does in fact support {}, but we don't use that style, so you have to get used to proper indentation.

**Comment by Shadowlight213** [comment]
@Robustin Ok. I made a pr to your branch. Should compile now.

**Comment by Shadowlight213** [comment]
@Robustin STOP MAKING MORE CHANGES, REQUIRING ME TO MAKE EVER MORE MERGE CONFLICT FIXES!!! REEEEEEEEEEEEEE

**Comment by Robustin** [comment]
wew, ok ok.

**Comment by Shadowlight213** [comment]
@Robustin Ok. everything fixed and it compiles locally for me.

**Comment by Robustin** [comment]
Woo green check mark. Thanks to @Shadowlight213 for hammering out the compiling issues. Not even the cult can wield whatever dark magicks are necessary to appease Travis.

File: code/game/gamemodes/cult/cult.dm (hunk 1 above)
Comment (KazeEspada) [inline code/game/gamemodes/cult/cult.dm]: We make a bigger nar nar rune and make it require more people.

**Comment by Bawhoppen** [comment]
:+1:x:100: praise our savior

**Comment by Robustin** [comment]
>40 person newcult round
>NarSie summoning begun 10 minutes into the round
>Anyone who comes near gets the ol' stun and instantly sacrificed at full HP meme, instantly becomes artificer to rinse and repeat this meme

Plz merge.

**Comment by Iamgoofball** [comment]
@Incoming5643

**Comment by Incoming5643** [comment]
No problems here from me, next time I do real work on datum antags (soon™.) it'll have to start with vigorous conflict resolution anyways, and I was going to throw out the little work I had done on cult stuff anyway

**Comment by Shadowlight213** [comment]
@Robustin you'll need to ask someone for a test merge on the server if you haven't already. While I did get the code to compile, I didn't have time to actually run it locally.
I would not recommend it being actually merged until it's tested.

**Comment by PKPenguin321** [comment]
merge conflicts reeee

**Comment by Shadowlight213** [comment]
I'm going to be expected to fix these conflicts, aren't I?
Sigh

**Comment by Shadowlight213** [comment]
Ok, I fixed the merge conflicts, but git freaked out with the commit log on pulling from master.
https://github.com/Robustin/-tg-station/pull/4

**Comment by Shadowlight213** [comment]
Travis actually technically passed. It failed on dreamstation for some reason.

**Comment by duncathan** [comment]
Gotcha

**Comment by Shadowlight213** [comment]
So. test merge on sybil. A few issues.
Cultists aren't assigned correctly at round start. while traitor panel shows them as cultists, they dont have any of the cultist gear, nor their antag hud. deculting and reculting them via traitor panel fixes that.
No end round report.
The cultist amulet paper doesn't have a spawn tome function. it seems to be for newcult.

**Comment by phil235** [comment]
>55 commits

Please squash your commits...

**Comment by Robustin** [comment]
Ugh, the amulet stuff was def. in the oldcult code I copied most of this from. I don't even have much experience myself with it. That's not a serious concern though as long as cultists get their tomes, another cultist can always summon a new tome if you lose yours. 

Fixing the assignment stuff is going to be a lot harder.

**Comment by Shadowlight213** [comment]
I'm fairly certain the HUDs not showing up means that they aren't getting their antag status properly assigned or something. From what I observed I don't think the cultists got their intro text either as they didn't seem to realize they were cult until I reapplied the status. However, they still were on the check antag list, as well as having cultist prayers and having the cultist antag flag in the traitor panel.
I'm guessing a setup cultist proc is failing to be called somewhere for the roundstart cultists beyond setting their special role to cultist and adding them to the cultist list.

File: code/game/gamemodes/cult/cult.dm [hunk 10]
```diff
@@ -75,18 +99,40 @@
...
+				else
+					explanation = "Free objective."
+			if("eldergod")
+				explanation = "Summon Nar-Sie via the rune 'Call Forth The Geometer'. It will only work if nine acolytes stand on and around it."
+		cult_mind.current << "<B>Objective #[obj_count]</B>: [explanation]"
+		cult_mind.memory += "<B>Objective #[obj_count]</B>: [explanation]<BR>"
+
 /datum/game_mode/cult/post_setup()
 	modePlayer += cult
-	for(var/datum/mind/cult_mind in cult)
```
Comment (Shadowlight213) [inline code/game/gamemodes/cult/cult.dm]: Why did you remove this code block? This is what gave the original cultists their hud, objectives, gear, etc.
Move the objective generation code above this.

Comment (Robustin) [inline code/game/gamemodes/cult/cult.dm]: There were about 1,000 lines to compare in addition to what I added, I guess it was an inadvertent cut. I'd go and check for sure but I'll just take your word for it.

Comment (Shadowlight213) [inline code/game/gamemodes/cult/cult.dm]: I already checked before I posted. :P

**Comment by Robustin** [comment]
Ran it locally, everything looked in order. Hud, items, etc. was there like normal and im happy to report the narsie rune works exactly as i had hoped.

**Comment by Core0verload** [comment]
Mergetest when?

**Comment by Shadowlight213** [comment]
Added some logging, as well as a delay for removing the narsie rune for cultists.
Fixed tome spawning from supply amulets and cultslur decreasing.
https://github.com/Robustin/-tg-station/pull/5

**Comment by Robustin** [comment]
Squashing commits has proven to be an intractable nightmare.

Merge and end this.

**Comment by KorPhaeron** [comment]
No need for two of the same PR
